.github/workflows/integration_tests.yaml

```diff
@@ -64,17 +69,135 @@ jobs:
             --token ${{secrets.E2E_TESTING_TOKEN}} \
             --token-alt ${{secrets.E2E_TESTING_TOKEN_ALT}} \
             tests/integration/test_charm_fork_repo.py
+  integration-test-charm-scheduled-events:
+    name: Integration test charm scheduled events
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          juju-channel: 3.1/stable
+          provider: lxd
+      - name: Enable br_netfilter
+        run: sudo modprobe br_netfilter
+      - name: Run Integration tests
+        run: |
+          tox -e integration -- \
+            --keep-models \
+            --path ${{secrets.E2E_TESTING_REPO}} \
+            --token ${{secrets.E2E_TESTING_TOKEN}} \
+            --token-alt ${{secrets.E2E_TESTING_TOKEN_ALT}} \
+            tests/integration/test_charm_scheduled_events.py
+  integration-test-charm-no-runner-juju-2-9:
+    name: Integration test charm with no runner with juju 2.9
```

Comment:
Would it be possible to use matrix for versions of juju we want to test against? E.g., at some point we might need to test 3.2, 3.1 and 2.9 etc

Reply:
Done.